pgroll-internals.mdx

```diff
@@ -0,0 +1,576 @@
+_pgroll_trigger_users_description BEFORE INSERT OR UPDATE ON users FOR EACH ROW EXECUTE FUNCTION _pgroll_trigger_users_description()
+```
+
+These triggers are used by `pgroll` to ensure that any values written into the old `description` column are copied over to the `_pgroll_new_description` column (rewriting values using the `up` SQL from the migration) and to copy values written to the `_pgroll_new_description` column back into the old `description` column (rewriting values using the`down` SQL from the migration). Our migration did not specify any `down` SQL, so the default behaviour is just to copy data from the `_pgroll_new_description` column into the `description` column without modification.
+
+Let's see the first of those triggers in action.
+
+First set the [search path](https://www.postgresql.org/docs/current/ddl-schemas.html#DDL-SCHEMAS-PATH) for your Postgres session to use the old schema:
+
+```sql
+SET search_path = 'public_01_create_users_table'
+```
+
+Now insert some data into the `users` table through the `users` view:
+
+```sql
+INSERT INTO users(name, description) VALUES ('Alice', 'this is Alice'), ('Bob', NULL)
+```
+
+This inserts two new users into the `users` table, one with a `description` and one without.
+
+Let's check that the data was inserted:
+
+```sql
+SELECT * FROM users WHERE name = 'Alice' or name = 'Bob'
+```
+
+Running this query should show:
+
+```
++--------+-------+---------------------+
+| id     | name  | description         |
++--------+-------+---------------------+
+| 100001 | Alice | this is Alice       |
+| 100002 | Bob   | NULL                |
++--------+-------+---------------------+
+```
+
+The trigger should have copied the data that was just written into the old `description` column (without the `NOT NULL` constraint) into the `_pgroll_new_description` column (with the `NOT NULL` constraint )using the `up` SQL from the migration.
+
+Let's check. Set the search path to the new version of the schema:
+
+```sql
+SET search_path = 'public_02_user_description_set_nullable'
+```
+
+and find the users we just inserted:
+
+```sql
+SELECT * FROM users WHERE name = 'Alice' or name = 'Bob'
+```
+
+The output should look like this:
+
+```
++--------+-------+---------------------+
+| id     | name  | description         |
++--------+-------+---------------------+
+| 100001 | Alice | this is Alice       |
+| 100002 | Bob   | description for Bob |
++--------+-------+---------------------+
+```
+
+Notice that the trigger installed by `pgroll` has rewritten the `NULL` value inserted into the old schema by using the `up` SQL from the migration definition.
+
+> How do applications configure which version of the schema to use?
+> `pgroll` allows old and new versions of an application to exist side-by-side during a migration. Each version of the application should be configured with the name of the correct version schema, so that the application sees the database schema that it expects.
```

Comment:
```suggestion
> How do applications configure which version of the schema to use?\
> `pgroll` allows old and new versions of an application to exist side-by-side during a migration. Each version of the application should be configured with the name of the correct version schema, so that the application sees the database schema that it expects.
```

Pretty sure this breaks the lines. Related to my comment above, maybe we choose one format for the Q&A call out boxes. Bold question, non-bold answer or italicized, etc. I love them by the way, they're pretty engaging.